Tests/Unit/NetworkServer/LnsProtocolMessageProcessorTests.cs

```diff
@@ -334,7 +323,7 @@ public async Task InternalHandleDataAsync_ShouldProperlyCreateLoraPayloadForUpdf
 
             // assert
             Assert.NotNull(loRaRequest);
-            Assert.True(AreRxpkEqual(loRaRequest.Rxpk, expectedRxpk));
+            Assert.True(RadioMetadata(loRaRequest.RadioMetadata, expectedRadioMetadata));
```

Comment:
Since you made `RadioMetadata` a record, you get equality comparison for free:

```suggestion
            Assert.Equal(loRaRequest.RadioMetadata, expectedRadioMetadata);
```